pkg/math/testutils.go

```diff
@@ -0,0 +1,51 @@
+package math
+
+import (
+	"math/big"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+// VerifyPrice verifies that the expected price matches the actual price within an acceptable delta.
+func VerifyPrice(
+	t *testing.T,
+	expected,
+	actual *big.Int,
+	acceptableDelta float64,
+) {
+	t.Helper()
+
+	zero := big.NewInt(0)
+	if expected.Cmp(zero) == 0 {
+		require.Equal(t, zero, actual)
+		return
+	}
+
+	var diff *big.Float
+	if expected.Cmp(actual) > 0 {
+		diff = new(big.Float).Sub(new(big.Float).SetInt(expected), new(big.Float).SetInt(actual))
+	} else {
+		diff = new(big.Float).Sub(new(big.Float).SetInt(actual), new(big.Float).SetInt(expected))
```

Comment:
Can't you just use the `Abs` function here?

Reply:
I don't fully remember why i didn't use `Abs` but i do remember not using it for a specific reason. This code might change as we convert to big.Floats so I'd prefer leaving it as is for now.